File: template/nestJs/libs/models/src/menu.ts
import { ApiProperty } from '@nestjs/swagger';
import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm';

@Entity('menu')
export class Menu {

  @ApiProperty({
    description: '菜单ID'
  })
  @PrimaryGeneratedColumn()
  id: number;

  @ApiProperty({
    description: '菜单名'
  })
  @Column()
  name: string;

  @ApiProperty({
    description: '菜单序列'
  })
  @Column()
  order: number;
  @ApiProperty({
    description: '菜单父级',
    nullable: true
  })
  @Column({ nullable: true })
  parentId: number;

  @ApiProperty({
    description: '菜单类型',
  })
  @Column()
  menuType: string;
  @ApiProperty({
    description: '菜单图标',
  })
  @Column({ nullable: true })
  icon: string;

  @ApiProperty({
    description: '菜单对应的前端组件',
  })
  @Column()
  component: string;

  @ApiProperty({
    description: '菜单路由地址',
  })
  @Column()
  path: string;

  @ApiProperty({
    description: '菜单国际化key',
  })
  @Column()
  locale: string;
}

File: template/springboot/src/main/java/com/TinyPro/entity/po/Menu.java
package com.TinyPro.entity.po;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.DynamicUpdate;

import java.io.Serializable;


@Entity
@Table(
        name = "menu",
        uniqueConstraints = @UniqueConstraint(
                name = "uk_menu_identity",
                columnNames = {
                        "name", "order", "menuType", "parentId",
                        "path", "icon", "component", "locale"
                }
        )
)
@Data
@DynamicUpdate
public class Menu implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column(nullable = false)
    private String name;
    @Column(name = "`order`")
    @TableField("'order'")
    private Integer order;
    @Column(name = "parentId")
    private Integer parentId;
    @Column(name = "menuType")
    private String menuType;
    @Column(name = "icon")
    private String icon;
    @Column(name = "component")
    private String component;
    @Column(name = "path")
    private String path;
    @Column(name = "locale")
    private String locale;
}
